homeassistant/components/wemo.py

```diff
@@ -125,43 +129,51 @@ def setup_url_for_address(host, port):
 
     devices = []
 
-    _LOGGER.debug("Scanning statically configured WeMo devices...")
-    for host, port in config.get(DOMAIN, {}).get(CONF_STATIC, []):
-        url = setup_url_for_address(host, port)
-
-        if not url:
-            _LOGGER.error(
-                'Unable to get description url for %s',
-                '{}:{}'.format(host, port) if port else host)
-            continue
-
-        try:
-            device = pywemo.discovery.device_from_description(url, None)
-        except (requests.exceptions.ConnectionError,
-                requests.exceptions.Timeout) as err:
-            _LOGGER.error('Unable to access %s (%s)', url, err)
-            continue
-
-        if not [d[1] for d in devices
-                if d[1].serialnumber == device.serialnumber]:
-            devices.append((url, device))
-
-    if config.get(DOMAIN, {}).get(CONF_DISCOVERY):
-        _LOGGER.debug("Scanning for WeMo devices...")
-        for device in pywemo.discover_devices():
+    async def async_discover_wemo_devices(hass, config):
+        """Run discovery in an async context so setup can complete quickly."""
+        _LOGGER.debug("Scanning statically configured WeMo devices...")
+        for host, port in config.get(DOMAIN, {}).get(CONF_STATIC, []):
+            url = setup_url_for_address(host, port)
```

Comment:
This function is doing non asyncio I/O. We're not allowed to do that in a coroutine that will run in the event loop. It would hold up the event loop.